datastream-server/src/test/java/com/linkedin/datastream/server/zk/TestZkAdapter.java

```diff
@@ -751,14 +755,66 @@ public void testZookeeperSessionExpiry() throws InterruptedException {
     Mockito.verify(adapter, Mockito.times(1)).onSessionExpired();
   }
 
+  @Test
+  public void testZookeeperLockAcquire() throws InterruptedException {
```

Comment:
Should we also add a test for dependency locks? Feel free to defer this to a later point in time or have me take it up.

Reply:
There is already an existing test for dependency lock. Please check and you can enhance it if you feel it does not cover everything.

Reply:
Oops didn't check the full file for tests. Thanks for pointing it out.